viewing_party/party.py

```diff
@@ -1,23 +1,190 @@
+    for movie in friends_watched:
+        for title in movie["watched"]:
+            if title in friend_movie:
+                continue
+            friend_movie.append(title)
+    
+    for title in user_movie:
+        if title in friend_movie:
+            friend_movie.remove(title)
+            
+    return friend_movie
 
         
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    #create user_only movie list
+    #create friends_only movie list
+    #compare user and friend movie lists: remove user movies from friend list = recom_movies
+    #comopare recom_movies to user host: remove any movies not in user host/subscription
+    
+    friend_rec = []
+    user_watched = user_data["watched"]
+    friends = user_data["friends"]
+
+    for friend in friends:
+        for movie in friend["watched"]:
+            if movie not in user_watched and movie["host"] in user_data["subscriptions"]:
+                friend_rec.append(movie)
+
+    return friend_rec
+    
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
+def get_new_rec_by_genre(user_data):
+    #find user most freq genre = fav_genre
+    while len(user_data["watched"]) > 0:
+        user_genre_movies = []
+        rec_genre_movies = []
+        freq_genre = {} 
+        for movie in user_data["watched"]:
+            genre = movie["genre"]
+            if genre not in freq_genre:
+                freq_genre[genre] = 1
+            else:
+                freq_genre[genre] += 1   
+        fav_genre = max(freq_genre, key=freq_genre.get)
+    
+    
+        #create user_movie list with only movies with fav_genre
+    
+        user_movie = user_data["watched"]
+        for movie in user_movie:
+            if movie["genre"] == fav_genre:
+                user_genre_movies.append(movie)
+            
+        #create friend_movie list with only movies with fav_genre
+                #compare user_movie list to friend movie list
+            #remove user_movies from friend movie list
+        #return friend list
+
+            
+        for friend in user_data["friends"]:
+            for movie in friend["watched"]:
+                if movie["genre"] == fav_genre and movie not in user_movie:
+                    rec_genre_movies.append(movie)
+        return rec_genre_movies
+    return user_data["watched"]
+
+def get_rec_from_favorites(user_data):
+    user_favs = []
+    friends_movies = []
+    
+    # while len(user_data["friends"][0]["watched"]) > 0:
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friends_movies.append(movie)
 
+    # while len(user_data["favorites"]) > 0:
+    for movie in user_data["favorites"]:
+        if movie not in friends_movies:
+            user_favs.append(movie)
+    return user_favs

```

Comment:
👍 Great work Ashanti! We could also use helper functions her to compere the friends watched list to the user watched list.